### PR TITLE
Api 3008

### PR DESCRIPTION
**Pull request by GutsyGusty**
@joshcom @wlmcewen 
Allowing 1-6 decimals of precision on Dates, a colon in the offset, and allowing Z for UTC time. Offset is now required. I cannot imagine that being an issue, but maybe.

### CODE REVIEW

**Comment by GutsyGusty** [comment]
Don't merge quite yet.

**Comment by GutsyGusty** [comment]
It actually looks like we cannot require the timezone info. Which is lame! :see_no_evil:

File: lib/sparkql/token.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ module Sparkql::Token
   DECIMAL = /^\-?[0-9]+\.[0-9]+/
   CHARACTER = /^'([^'\\]*(\\.[^'\\]*)*)'/
   DATE = /^[0-9]{4}\-[0-9]{2}\-[0-9]{2}/
-  DATETIME = /^[0-9]{4}\-[0-9]{2}\-[0-9]{2}T[0-9]{2}\:[0-9]{2}\:[0-9]{2}(\.[0-9]{6})?((\+|-)[0-9]{4})?/
```
Comment (wlmcewen) [inline lib/sparkql/token.rb]: One thing that we should at least check generally, is if the old suffix format was ever used. I think it generally is convertible to the new one with the colon.

**Comment by wlmcewen** [comment]
Providing and documenting reasonable defaults seems like the way to go when the timezone info isn't provided. The rest looks good with one suggestion.

File: lib/sparkql/token.rb (hunk 1 above)
Comment (GutsyGusty) [inline lib/sparkql/token.rb]: Notice that the `:` is optional now to allow 4 numbers in a row as well. That is covered by other tests.

Comment (wlmcewen) [inline lib/sparkql/token.rb]: My mistake, I missed the `?` and somehow that it was also in the tests :+1:

**Pull request merged**